File: Programmers/level1/DeleteSameNumber.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    int temp = -1;
    vector<int> answer;

    for(int i=0; i<arr.size(); i++){
        if(arr[i] == temp) continue;
        temp = arr[i];
        answer.push_back(arr[i]);
    }

    return answer;
}

File: baek/Other/Delete_Same_Number.java
import java.util.*;

public class Delete_Same_Number {
	public int[] solution(int []arr) {
        ArrayList<Integer> list = new ArrayList<Integer>();
        int point = -1;
        for(int i:arr){
            if(point == i){
                continue;
            }
            point = i;
            list.add(i);
        }
        int[] answer = new int[list.size()];
        Iterator itr = list.iterator();
        for(int i=0; i< answer.length; i++){
            answer[i] = (int) itr.next();
        }
        return answer;
        
        
	}
}